framework/include/base/ChainControlDataSystem.h

```diff
@@ -0,0 +1,134 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ChainControlData.h"
+#include "MooseError.h"
+#include "MooseUtils.h"
+
+class ChainControl;
+
+/**
+ * System that manages ChainControls.
+ *
+ * To be owned by the MooseApp.
+ */
+class ChainControlDataSystem
+{
+public:
+  ChainControlDataSystem(MooseApp & app) : _app(app) {}
+
+  /**
+   * Queries if the chain control data of the given name exists.
+   *
+   * @param[in] data_name  Chain control data name
+   * @return True if the chain control data of the given name exists
+   */
+  bool hasChainControlData(const std::string & data_name) const
+  {
+    return _chain_control_data_map.find(data_name) != _chain_control_data_map.end();
```

Comment:
I would put all these methods as inline methods outside of the class definition. Want perusal of the class definition to be quick